app/src/main/java/com/example/feelgoodinc/services/CalendarService.java

```diff
@@ -0,0 +1,265 @@
+                                    calendarView.setCurrentDayBackgroundColor(colour);
+                                }
+                            }
+                            if (mood.getMoodType().equals(Mood.MoodType.AWFUL)) {
+                                int colour = ContextCompat.getColor(context, R.color.awful);
+                                addDateColourWithData(calendarView, mood.getMoodWhen(), journal.getContent(), colour);
+                                if(areDatesOnSameDay(mood.getMoodWhen(),Calendar.getInstance().getTime())){
+                                    calendarView.setCurrentDayBackgroundColor(colour);
+                                }
+
+                            }
+
+                        }
+                    }
+                });
+            });
+
+        }
+    }
+
+    /**
+     * Unbinds the {@link JournalService} and {@link MoodService} from the app
+     */
+    public void unBindServices(){
+        context.unbindService(serviceConnection1);
+        context.unbindService(serviceConnection2);
+    }
+
+    /**
+     * Checks if two dates are on the same day
+     * @param date1 the first {@link Date) object
+     * @param date2 the second {@link Date} object
+     * @return True if the dates are on the same day, false if otherwise
+     */
+    public static boolean areDatesOnSameDay(Date date1, Date date2) {
+        Calendar cal1 = Calendar.getInstance();
+        cal1.setTime(date1);
+
+        Calendar cal2 = Calendar.getInstance();
+        cal2.setTime(date2);
+
+        return cal1.get(Calendar.YEAR) == cal2.get(Calendar.YEAR) &&
+                cal1.get(Calendar.MONTH) == cal2.get(Calendar.MONTH) &&
+                cal1.get(Calendar.DAY_OF_MONTH) == cal2.get(Calendar.DAY_OF_MONTH);
+    }
+
+    /**
+     * Finds the journal in a list on a given date
+     * @param journals the list of journals
+     * @param date the date of creation
+     * @return {@link Journal} created on date
+     */
+    private Journal findJournal(List<Journal> journals, Date date){
+        for(Journal journal : journals) {
+            if(areDatesOnSameDay(journal.getCreatedWhen(),date)) {
+                return journal;
+            }
+        }
+        return null;
+    }
+
+    /**
+     * The service connection used to bind the {@link JournalService}
+     */
+    final ServiceConnection serviceConnection1 = new ServiceConnection() {
+        /**
+         * This binds the {@link JournalService} to the app and starts the binding to the {@link MoodService} on success
+         * @param componentName The concrete component name of the service that has
+         * been connected.
+         *
+         * @param iBinder The IBinder of the Service's communication channel,
+         * which you can now make calls on.
+         */
+        @Override
+        public void onServiceConnected(ComponentName componentName, IBinder iBinder) {
```

Comment:
Could have passed the type of service here as an argument, to avoid duplication. Doesn't really matter either way though